ddprof-lib/src/main/cpp/symbols_linux.cpp

```diff
@@ -418,6 +456,64 @@ void Symbols::parseKernelSymbols(CodeCache* cc) {
     fclose(f);
 }
 
+
+// Signal handler safe read mechanism
+// This is added to secure concurrent unloading of mappings
+// This should not happen in a pure Java environment,
+// however, with other native agents, we could have mappings
+// that unload while we are processing the mappings
+namespace {
+// Original SIGSEGV handler
+static struct sigaction original_sa;
+
+static void (*orig_segvHandler)(int signo, siginfo_t* siginfo, void* ucontext) = nullptr;
+
+// Simplified version of the unwinding's crash handler
+bool crashHandler(int signo, siginfo_t* siginfo, void* ucontext) {
+    StackFrame frame(ucontext);
+    uintptr_t pc = frame.pc();
+
+    uintptr_t length = SafeAccess::skipLoad(pc);
+    if (length > 0) {
+        // Skip the fault instruction, as if it successfully loaded NULL
+        frame.pc() += length;
+        frame.retval() = 0;
+        return true;
+    }
+
+    length = SafeAccess::skipLoadArg(pc);
+    if (length > 0) {
+        // Act as if the load returned default_value argument
+        frame.pc() += length;
+        frame.retval() = frame.arg1();
+        return true;
+    }
+
+    return false;
+}
+
+void segvHandler(int signo, siginfo_t* siginfo, void* ucontext) {
+    if (!crashHandler(signo, siginfo, ucontext) && orig_segvHandler) {
+        orig_segvHandler(signo, siginfo, ucontext);
+    }
+}
+}
+
+void Symbols::setupSignalHandlers() {
+    // Replacing the segvhandler only while parsing the libraries
+    // It can be unsafe to touch signal handlers in some Java distributions
+    // The assumption is that doing so for a short time is safe
+    // TODO: verify this assumption
+    orig_segvHandler = OS::replaceSigsegvHandler(segvHandler);
```

Comment:
here I removed some of the checks that were available within profiler.cpp on signal handlers.
Is my assumption LGTM ?